src/core/derivation.ts

```diff
@@ -119,23 +119,24 @@ export function trackDerivedFunction<T>(derivation: IDerivation, f: () => T) {
 	derivation.runId = ++globalState.runId;
 	const prevTracking = globalState.trackingDerivation;
 	globalState.trackingDerivation = derivation;
-	let hasException = true;
+	let hasException = false;
 	let result: T;
 	try {
 		result = f.call(derivation);
-		hasException = false;
+	} catch(e){
+		hasException = true;
+		handleExceptionInDerivation(derivation, e);
+		throw e;
```

Comment:
what @spion said ... 
what do you think about a test to assert the original stack trace?